test/tests/kernels/hfem/array_dirichlet_pjfnk.i

```diff
@@ -0,0 +1,120 @@
+[Mesh]
+  [square]
+    type = GeneratedMeshGenerator
+    nx = 3
+    ny = 3
+    dim = 2
+  []
+  build_all_side_lowerd_mesh = true
+[]
+
+[Variables]
+  [u]
+    order = THIRD
+    family = MONOMIAL
+    block = 0
+    components = 2
+  []
+  [lambda]
+    order = CONSTANT
+    family = MONOMIAL
+    block = INTERNAL_SIDE_LOWERD_SUBDOMAIN
+    components = 2
+  []
+  [lambdab]
+    order = CONSTANT
+    family = MONOMIAL
+    block = BOUNDARY_SIDE_LOWERD_SUBDOMAIN
+    components = 2
+  []
+[]
+
+[AuxVariables]
+  [v]
+    order = CONSTANT
+    family = MONOMIAL
+    block = 0
+    initial_condition = '1'
+  []
+[]
+
+[Kernels]
+  [diff]
+    type = ArrayDiffusion
+    variable = u
+    block = 0
+    diffusion_coefficient = dc
+  []
+  [reaction]
```

Comment:
Quite note: this kernel is added than the other input `array_dirichlet.i` is because otherwise the block diagonal matrix of `u` is only the stiffness matrix and we know stiffness matrix without any bc is singular. Adding reaction kernel (mass matrix) makes the block diagonal matrix of `u` non-singular. In `Executioner` we removed the `NEWTON` solve type.